### PR TITLE
Plugin folder

### PR DESCRIPTION
**Pull request by Peter9192**
This PR restructures the repository such that all model-specific things are grouped together in a model folder. 

refs #338

To do:

- [x] Also group tests per model instead of per type
- [x] Also use plugins for forcing and example parameter sets - ban all imports from plugin folder in rest of ewatercycle package (so it doesn't break if you extract all plugins to their own repos)

### CODE REVIEW

File: src/ewatercycle/defaults/bmi_handles.py
Comment (sverhoeven) [inline src/ewatercycle/defaults/bmi_handles.py:1]: BmiHandles is not being used. Remove or use?

**Comment by Peter9192** [comment]
We completely refactored parameter sets. We created a new folder called base where we can store all the pydantic base classes. New todo:

- [ ] Also move models, forcing etc. to `/base`
- [ ] Make sure CI is green again
- [ ] Update docs

**Comment by BSchilperoort** [comment]
@Peter9192 the sonarcloud analysis fails on this PR as it only checks the diff lines. Do we still want to fix some of these issues? Or just merge.

**Comment by Peter9192** [comment]
> @Peter9192 the sonarcloud analysis fails on this PR as it only checks the diff lines. Do we still want to fix some of these issues? Or just merge.

Let's merge

**Review by BSchilperoort — APPROVED** [review]
:rocket:  :smile:

**Pull request merged**